aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -579,23 +580,9 @@ bool canSliceViewTensor(const Tensor& src, MPSShape *mpsShape) {
                                                    name: nil];
     MPSGraphTensor *inputTensor = cachedGraph->inputTensor;
 
-    // Workaround for bool scatter/gather deficiency
-    // See https://github.com/pytorch/pytorch/issues/82663
-    if (needsBoolCast) {
```

Comment:
Why we don't need casting before calling asStridedLayer_pattern() for Macos 12?

Reply:
If the input/updates are bool or uint8_t, these are directly passed as int8 to the graph, so on macOS 12, the input passed to the graph will never be of bool data type